src/lightning_app/cli/lightning_cli_delete.py

```diff
@@ -56,7 +56,6 @@ def _find_cluster_for_user(app_name: str, cluster_id: Optional[str]) -> str:
         valid_cluster_list.append(cluster.id)
         if cluster.spec.cluster_type == V1ClusterType.GLOBAL and default_cluster is None:
             default_cluster = cluster.id
-            break
```

Comment:
Can you explain this change? Isn't this supposed to find the first (global) cluster in the list and choose it as default?
Once we have found one, we can break out of the loop.

Reply:
> Once we have found one, we can break out of the loop.

Only if we were looking for the default to begin with. With the `break` in there, the `valid_cluster_list` doesn't get properly populated, so the test `test_find_cluster_for_user_when_provided_valid_cluster_id` fails

Reply:
aha that was my dismissed concer in https://github.com/Lightning-AI/lightning/pull/15783#discussion_r1039751635 :D